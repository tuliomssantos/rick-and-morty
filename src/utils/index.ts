export * from './createEmotionCache'
export * from './renderWithTeme'
