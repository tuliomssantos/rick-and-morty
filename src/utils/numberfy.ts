export interface INumberfy {
  (value: string | string[] | undefined, defaultValue: number): number
}

export const numberfy: INumberfy = (value, defaultValue) => {
  if (value) {
    return Number(value)
  }

  return defaultValue
}
