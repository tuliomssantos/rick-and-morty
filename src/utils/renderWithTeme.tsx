import { render, RenderResult } from '@testing-library/react'

import { CustomThemeProvider } from '../styles/theme'

import { QueryClient, QueryClientProvider } from 'react-query'

// import {setLogger} from 'react-query'
// setLogger({
//   log: console.log,
//   warn: console.warn,
//   // âœ… no more errors on the console
//   // eslint-disable-next-line @typescript-eslint/no-empty-function
//   error: () => {},
// })

const createTestQueryClient = () =>
  new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  })

export const renderWithTheme = (children: React.ReactNode): RenderResult => {
  const testQueryClient = createTestQueryClient()

  const { rerender, ...result } = render(
    <QueryClientProvider client={testQueryClient}>
      <CustomThemeProvider>{children}</CustomThemeProvider>
    </QueryClientProvider>
  )

  return {
    ...result,
    rerender: (rerenderChildren: React.ReactNode) =>
      rerender(
        <QueryClientProvider client={testQueryClient}>
          <CustomThemeProvider>{rerenderChildren}</CustomThemeProvider>
        </QueryClientProvider>
      ),
  }

  // return render(
  //   <QueryClientProvider client={queryClient}>
  //     <CustomThemeProvider>{children}</CustomThemeProvider>
  //   </QueryClientProvider>
  // )
}
