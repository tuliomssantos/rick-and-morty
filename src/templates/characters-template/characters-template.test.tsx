import { renderWithTheme } from '@/src/utils/renderWithTeme'

import { screen } from '@testing-library/react'

import CharactersTemplate from '.'

describe('<CharactersTemplate />', () => {
  it('should find Rick Sanchez on page 1', async () => {
    renderWithTheme(<CharactersTemplate />)

    expect(
      await screen.findByRole('heading', { name: /rick sanchez/i })
    ).toBeInTheDocument()
  })

  it('should not display loading on first render', async () => {
    renderWithTheme(<CharactersTemplate />)

    expect(screen.queryByTestId('rectangular-skeleton')).toBeNull()
  })
})
