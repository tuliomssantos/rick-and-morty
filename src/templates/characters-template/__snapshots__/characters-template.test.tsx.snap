// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharactersTemplate /> should render correctly 1`] = `
<div
  class="css-j8exxa"
>
  <div
    class="css-15wsh61"
  >
    <div
      class="css-w6gvpm"
      data-testid="header"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-18bxxyv-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1tg7ukt-MuiToolbar-root"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-thg7e2-MuiButtonBase-root-MuiIconButton-root"
            data-testid="menu-button"
            tabindex="0"
            type="button"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDgiIGhlaWdodD0iNDAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Logo"
                data-nimg="intrinsic"
                decoding="async"
                src="/_next/image?url=%2Fimages%2Fportal.png&w=96&q=75"
                srcset="/_next/image?url=%2Fimages%2Fportal.png&w=48&q=75 1x, /_next/image?url=%2Fimages%2Fportal.png&w=96&q=75 2x"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root css-1mid3pv-MuiButtonBase-root"
            tabindex="0"
            type="button"
          >
            <div
              class="css-1aekqjx"
              data-testid="logo"
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
              >
                <img
                  alt="Logo"
                  data-nimg="fill"
                  decoding="async"
                  sizes="100vw"
                  src="/_next/image?url=%2Fimages%2Frick_and_morty_logo.svg&w=3840&q=75"
                  srcset="/_next/image?url=%2Fimages%2Frick_and_morty_logo.svg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Frick_and_morty_logo.svg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Frick_and_morty_logo.svg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Frick_and_morty_logo.svg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Frick_and_morty_logo.svg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Frick_and_morty_logo.svg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Frick_and_morty_logo.svg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Frick_and_morty_logo.svg&w=3840&q=75 3840w"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </div>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </header>
    </div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1s5cpj8-MuiContainer-root"
    >
      <div
        class="css-0"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd css-1wssyo0-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1iiaq29-MuiInputAdornment-root"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="SearchIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </div>
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd css-vxvc7x-MuiInputBase-input-MuiOutlinedInput-input"
              id="input-with-icon-textfield"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-9425fu-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-nnbavb"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="css-eu7k4z"
      />
      <div
        class="css-b63z21"
      >
        <nav
          aria-label="pagination navigation"
          class="MuiPagination-root MuiPagination-text css-1oj2twp-MuiPagination-root"
        >
          <ul
            class="MuiPagination-ul css-wjh20t-MuiPagination-ul"
          >
            <li>
              <button
                aria-label="Go to previous page"
                class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-disabled MuiPaginationItem-previousNext css-6y92bt-MuiButtonBase-root-MuiPaginationItem-root"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                  data-testid="NavigateBeforeIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                  />
                </svg>
              </button>
            </li>
            <li>
              <button
                aria-current="true"
                aria-label="page 1"
                class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-selected MuiPaginationItem-page css-6y92bt-MuiButtonBase-root-MuiPaginationItem-root"
                tabindex="0"
                type="button"
              >
                1
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </li>
            <li>
              <button
                aria-label="Go to next page"
                class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-disabled MuiPaginationItem-previousNext css-6y92bt-MuiButtonBase-root-MuiPaginationItem-root"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                  data-testid="NavigateNextIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                  />
                </svg>
              </button>
            </li>
          </ul>
        </nav>
      </div>
    </div>
    <footer
      class="css-kutwh4"
      data-testid="footer"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-camo0u-MuiContainer-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        >
          Made with ❤️ by 
          <a
            class="css-ksdukt"
            href="https://github.com/tuliomssantos"
          >
            tuliomssantos
          </a>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        >
          For educational purposes only
        </p>
      </div>
    </footer>
  </div>
</div>
`;
